### PR TITLE
feat: extend GHA integration to support `pull_request_target` events

### LINKED ISSUE #331: Extend GHA integration to support `pull_request_target` events
## Overview

**Is your feature request related to a problem? Please describe.**

The current GitHub Action (GHA) integration is limited to `pull_request` events only. It does not support pull requests from forked repos (e.g., `pull_request_target` events). This was considered acceptable as the [Phylum GitHub App](https://github.com/marketplace/phylum-io) was promoted as the primary option for GitHub users. However, the _action_ now supports manifests in the form of lockfile generation while the app does not. That means more users are going to turn to the action and will want it to work in more environments than simply internal pull requests.

**Describe the solution you'd like**

Add support to the GHA for PRs, regardless of where they originate.

**Describe alternatives you've considered**

N/A

## Additional context

The `pull_request_target` event [appears to have the same webhook event payload](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target) as the `pull_request` event. This _should_ allow for simple inclusion, but further investigation and testing is needed to uncover any differences.

## Acceptance criteria

- [x] `pull_request_target` events are supported
- [ ] Documentation is updated

### PR DESCRIPTION
**Pull request by maxrake**
feat!: extend GHA integration to support `pull_request_target` events

This change makes it possible to use the GitHub Action (GHA) integration
for PRs that originate from forks, by enabling workflows to run on the
`pull_request_target` event.

This change comes with some heavy security considerations. In order for
the GHA to run, a workflow using it will need to checkout the PRs code.
This requires very careful attention so as not to allow so-called
"pwn requests":

* https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
* https://blog.gitguardian.com/github-actions-security-cheat-sheet/

To that end, the documentation updates that go with this change will be
explicit in their warnings about how to properly use the Phylum GHA in
the most secure way possible. This effectively means the workflow should
be limited to executing the bare minimum required for Phylum analysis.

Further, the Phylum analysis that does run will need to do so in a way
that does not allow for arbitrary code execution. Therefore, this change
also makes use of the Phylum CLI feature to disable lockfile generation
for untrusted contexts like the `pull_request_target` event trigger in
the GitHub Actions platform that this PR is looking to support.

Unfortunately, this approach does not allow for repositories with only
manifests (e.g., libraries). That is, manifests without a corresponding
lockfile. Those will fail to parse, with a new return code unique to
that situation. The only recourse, which is given in a log message and
will be spelled out more explicitly in the corresponding documentation
updates, is to reccomend adding a lockfile instead of or along with the
manifest.

Additional changes made include:

* Add predicate for if Phylum sandbox will work in current environment
* Create a `CLIExitCode` enumeration to track Phylum CLI exit codes
* Update `ReturnCode` enumeration for tracking `phylum-ci` return codes
  * Add descriptions
  * Make a distinction between _policy_ failures and general failures
  * Add new code for when no dependency files were provided or detected
  * Add new code for when a manifest is attempted to be parsed but
    lockfile generation has been disabled
* Ensure lockfiles provided as arguments are only filtered once
* Use `phylum project status` instead of `phylum project list` for
  setting the repository URL
* Update unit tests
* Format and refactor throughout
  * Show the final return code, _after_ any modifications

Closes #331

BREAKING CHANGE: Phylum CLI installs before v5.9.0-rc2 are no longer
supported. A version with support for disabling lockfile generation and
skipping sandbox usage is required.

BREAKING CHANGE: The `phylum-ci` return code for a policy violation that
results from a Phylum analysis has been changed from 1 to 2 in order to
make it distinct from the default failure code that is generated for all
raised `SystemExit` exceptions with a message instead of a code.

## Checklist

- [x] Does this PR have an associated issue (i.e., `closes #<issueNum>` in description above)?
- [x] Have you ensured that you have met the expected acceptance criteria?
- [x] Have you created sufficient tests?
  - Manual testing was performed with a fork of a personal repo
- [ ] Have you updated all affected documentation?
  - https://github.com/phylum-dev/phylum-analyze-pr-action/pull/32
    - (Still need to update that PR)

## Testing

Changes from this PR have been built in Docker images and hosted [on Docker Hub for my account](https://hub.docker.com/r/maxrake/phylum-ci/tags), as the `gh_pr_target` tag.

Screenshots from the manual testing may be added here soon.

### CODE REVIEW

File: src/phylum/ci/ci_github.py
Comment (maxrake) [inline src/phylum/ci/ci_github.py:1]: Commenting here as a way to remember to keep this PR open until the Phylum CLI changes are made...

The minimum Phylum CLI version constants will need to be updated in `src/phylum/constants.py` once there is a release out containing the lockfile generation security hardening.

**Comment by maxrake** [comment]
The Dockerfile tests are going to fail until a non-release candidate of the Phylum CLI is released with a version greater than v5.9.0-rc2

**Review by cd-work — CHANGES_REQUESTED** [review]

File: src/phylum/ci/ci_base.py [hunk 1]
```diff
@@ -58,6 +59,7 @@ def __init__(self, args: Namespace) -> None:
         self._returncode = ReturnCode.SUCCESS
         self._analysis_report = "No analysis output yet"
         self.ci_platform_name = "Unknown"
+        self.no_gen = False  # AKA disable_lockfile_generation
```
[the remark this answers is not in the document]
Comment (maxrake) [reply src/phylum/ci/ci_base.py]: Fair. It was a feeble attempt to keep from expanding a handful of lines into multiple lines each. It has been changed.

File: src/phylum/ci/ci_github.py
Comment (maxrake) [reply src/phylum/ci/ci_github.py:1]: The CLI changes were made and included in the v5.9.0 release. The Dockerfile tests are passing again.

**Pull request merged**